src/main/tut/1.9-function-types.md

```diff
@@ -0,0 +1,143 @@
+```Haskell
+eval :: ((a⇒b) × a) -> b
+```
+```scala
+def eval[`A=>B`, A, B]: ((`A=>B`, A)) => B
+```
+................
+```Haskell
+g :: z × a -> b
+```
+```scala
+def g[Z, A, B]: ((Z, A)) => B
+```
+................
+```Haskell
+h :: z -> (a⇒b)
+```
+```scala
+def h[Z, `A=>B`]: Z => `A=>B`
+```
+................
+```Haskell
+a -> (b -> c)
+```
+```scala
+A => (B => C)
+```
+................
+```Haskell
+a -> b -> c
+```
+```scala
+A => B => C
+```
+................
+```Haskell
+catstr :: String -> String -> String
+catstr s s’ = s ++ s’
+```
+```tut:silent
+val catstr: (String, String) => String = {
+  case (s, sᛌ) => s ++ sᛌ
```

Comment:
the case is not necessary here, you're taking two arguments, not a pair (in scala these two concepts aren't the same):
```scala
  val catstr: (String, String) => String = (s, sᛌ) => s ++ sᛌ
```

That said, the book is talking specifically about currying here, so this should probably be curried `val catstr: String => String => String`

I'd also recommend picking something less confusing then `sᛌ`. It' easy to confuse it with `s'` which won't compile in Scala

Reply:
Followed @kailuowang suggestion for prime symbol in readme. Should it be removed from code and readme?

Reply:
oops, my bad then :) Leave it as is